model/patch/patch.go

```diff
@@ -138,32 +138,33 @@ type GitMetadata struct {
 type Patch struct {
-	Id              mgobson.ObjectId       `bson:"_id,omitempty"`
-	Description     string                 `bson:"desc"`
-	Path            string                 `bson:"path,omitempty"`
-	Project         string                 `bson:"branch"`
-	Githash         string                 `bson:"githash"`
-	PatchNumber     int                    `bson:"patch_number"`
-	Author          string                 `bson:"author"`
-	Version         string                 `bson:"version"`
-	Status          string                 `bson:"status"`
-	CreateTime      time.Time              `bson:"create_time"`
-	StartTime       time.Time              `bson:"start_time"`
-	FinishTime      time.Time              `bson:"finish_time"`
-	BuildVariants   []string               `bson:"build_variants"`
-	Tasks           []string               `bson:"tasks"`
-	VariantsTasks   []VariantTasks         `bson:"variants_tasks"`
-	SyncAtEndOpts   SyncAtEndOptions       `bson:"sync_at_end_opts,omitempty"`
-	Patches         []ModulePatch          `bson:"patches"`
-	Parameters      []Parameter            `bson:"parameters,omitempty"`
-	Activated       bool                   `bson:"activated"`
-	PatchedConfig   string                 `bson:"patched_config"`
-	Alias           string                 `bson:"alias"`
-	Triggers        TriggerInfo            `bson:"triggers"`
-	BackportOf      BackportInfo           `bson:"backport_of,omitempty"`
-	MergePatch      string                 `bson:"merge_patch"`
-	GithubPatchData thirdparty.GithubPatch `bson:"github_patch_data,omitempty"`
-	GitInfo         *GitMetadata           `bson:"git_info,omitempty"`
+	Id                   mgobson.ObjectId       `bson:"_id,omitempty"`
+	Description          string                 `bson:"desc"`
+	Path                 string                 `bson:"path,omitempty"`
+	Project              string                 `bson:"branch"`
+	Githash              string                 `bson:"githash"`
+	PatchNumber          int                    `bson:"patch_number"`
+	Author               string                 `bson:"author"`
+	Version              string                 `bson:"version"`
+	Status               string                 `bson:"status"`
+	CreateTime           time.Time              `bson:"create_time"`
+	StartTime            time.Time              `bson:"start_time"`
+	FinishTime           time.Time              `bson:"finish_time"`
+	BuildVariants        []string               `bson:"build_variants"`
+	Tasks                []string               `bson:"tasks"`
+	VariantsTasks        []VariantTasks         `bson:"variants_tasks"`
+	SyncAtEndOpts        SyncAtEndOptions       `bson:"sync_at_end_opts,omitempty"`
+	Patches              []ModulePatch          `bson:"patches"`
+	Parameters           []Parameter            `bson:"parameters,omitempty"`
+	Activated            bool                   `bson:"activated"`
+	PatchedParserProject string                 `bson:"patched_parser_project"`
```

Comment:
So unfortunately we can't change the bson tag of an existing variable, because this already exists as `patched_config` in the database, and now we'll have no way to read any existing patch document into the patch struct.

We could still just change the variable name, so that in the code we have `PatchedParserProject` and `PatchedProjectConfig`, but in the DB have `patched_config` and `patched_project_config` or we can leave the variable name as PatchedConfig (for consistency with the bson tag) and use a different variable for the new one. There are truly pros and cons to either but whichever we choose, we should add a comment here to distinguish the two.

Reply:
I've gone ahead and reverted `patched_parser_project` back to `patched_config` and using `patched_project_config` for the new project configs, while keeping the struct name as `PatchedParserProject ` although it deviates from the BSON I do think it makes the code more legible that way.